science/plumed/Portfile

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+
+name                plumed
+PortGroup           mpi 1.0
+PortGroup           linear_algebra 1.0
+PortGroup           debug 1.0
+
+version             2.3.1
+categories          science
+
+# Most of the PLUMED code is L-GPL3. However, PLUMED containts
+# molfile plugins from VMD that are released with a BSD-like license
+# http://www.ks.uiuc.edu/Research/vmd/plugins/molfile/
+license             LGPL-3 BSD
+maintainers         gmail.com:giovanni.bussi openmaintainer
+description         PLUMED is a plugin for molecular dynamics
+long_description    PLUMED is a plugin for molecular dynamics that can be used \
+                    in combination with popular molecular dynamics codes to perform biased simulations. \
+                    Additionally, it can be used as a standalone tool to analyze trajectories.
+
+platforms           darwin
+
+homepage            http://www.plumed.org/
+
+master_sites        https://github.com/plumed/plumed2/releases/download/v${version}/
+distfiles           plumed-src-${version}.tgz
+
+checksums \
+   plumed-src-${version}.tgz \
+          sha256 0f0ec4998bd5fe9a897fdbecd26efb14859179bc7e9b43cb2c457c33cc64d5cf \
+          rmd160 569af66d66999a17e04c2912251a2135c7ee4f93
+
+# This is required so that patch made with "git format-patch" works correctly
+patch.pre_args -p1
+# In case patches from github should be included to fix behavior on macports one should
+# - put patch files (as obtained with git format-patch) in files/ directory
+# - list their name here:
+patchfiles 0001-Added-enable-libsearch-to-configure.patch
+
+# Disable additional features.
+# They are then re-enabled when selecting proper variants
+# this is important for features that require an additional package to be
+# sure that even if the user has that package already installed plumed is
+# not going to use it.
+# Also avoid searching for doxygen.
+configure.args-append \
+               --disable-doc \
+               --disable-libsearch \
+               --disable-mpi 
+
+
+# Hardcode path for libplumedKernel.dylib.
+# This allows to patch MD codes using the --runtime option but using as
+# default kernel the installed one. In this way, MacPorts users
+# can just use patched MD codes with the installed plumed or replace it
+# by setting PLUMED_KERNEL at runtime
+configure.cppflags-append "-D__PLUMED_DEFAULT_KERNEL=${prefix}/lib/libplumedKernel.dylib"
+
+compilers.choose    cc cxx
+mpi.setup
+
+if {[mpi_variant_isset]} {
+  configure.args-replace --disable-mpi --enable-mpi
+}
+
+# libraries
+configure.ldflags-append -lmatheval -lxdrfile -lz -lgsl
```

Comment:
The build system should add those flags automatically.
Read as: the build system should be fixed upstream (it is acceptable to use such workarounds until this gets fixed, but it should really really be fixed eventually).